src/services/idleActionService.js

```diff
@@ -0,0 +1,44 @@
+class IdleActionService {
+  constructor(conn) {
+    this.conn = conn;
+  }
+
+  setIdleStatus(heroId, type) {
+    if (Number.isNaN(Number(heroId))) return Promise.reject(new Error(400));
+    if (type !== ('rest' || 'scout' || 'train')) {
+      return Promise.reject(new Error(400));
+    }
+    const currentTime = new Date() / 1000;
+    let query = 'SELECT * FROM idleStatus WHERE heroId = ?;';
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [heroId, type], (err, row) => {
```

Comment:
i dont see a second questionmark in the query, do you need the type in the input?

Reply:
corrected